crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -366,63 +359,23 @@ impl PackageKind {
 
 #[cfg(test)]
 mod tests {
-    use ruff_db::file_system::{FileSystemPath, FileSystemPathBuf};
+    use ruff_db::file_system::FileSystemPath;
     use ruff_db::vfs::{system_path_to_file, VfsFile, VfsPath};
 
-    use crate::db::tests::TestDb;
-    use crate::module::{ModuleKind, ModuleName};
-
-    use super::{
-        path_to_module, resolve_module, set_module_resolution_settings, ModuleResolutionSettings,
-        TYPESHED_STDLIB_DIRECTORY,
-    };
-
-    struct TestCase {
-        db: TestDb,
-
-        src: FileSystemPathBuf,
-        custom_typeshed: FileSystemPathBuf,
-        site_packages: FileSystemPathBuf,
-    }
-
-    fn create_resolver() -> std::io::Result<TestCase> {
-        let mut db = TestDb::new();
-
-        let src = FileSystemPath::new("src").to_path_buf();
-        let site_packages = FileSystemPath::new("site_packages").to_path_buf();
-        let custom_typeshed = FileSystemPath::new("typeshed").to_path_buf();
-
-        let fs = db.memory_file_system();
-
-        fs.create_directory_all(&src)?;
-        fs.create_directory_all(&site_packages)?;
-        fs.create_directory_all(&custom_typeshed)?;
+    use crate::db::tests::{create_resolver_builder, TestCase};
+    use crate::module::ModuleKind;
+    use crate::module_name::ModuleName;
 
-        let settings = ModuleResolutionSettings {
-            extra_paths: vec![],
-            workspace_root: src.clone(),
-            site_packages: Some(site_packages.clone()),
-            custom_typeshed: Some(custom_typeshed.clone()),
-        };
-
-        set_module_resolution_settings(&mut db, settings);
-
-        Ok(TestCase {
-            db,
-            src,
-            custom_typeshed,
-            site_packages,
-        })
-    }
+    use super::*;
 
     #[test]
     fn first_party_module() -> anyhow::Result<()> {
-        let TestCase { db, src, .. } = create_resolver()?;
+        let TestCase { db, src, .. } = create_resolver_builder()?.build();
```

Comment:
Nit: Maybe consider adding a simple `setup_resolver_test()` function that calls `create_resolver_builder().build()` to make the tests a bit less verbose.